data/src/config/server.rs

```diff
@@ -99,10 +101,44 @@ impl Server {
             connection::Security::Unsecured
         };
 
+        let proxy = self.proxy.clone().map(|p| p.into());
+
         connection::Config {
             server: &self.server,
             port: self.port,
             security,
+            proxy,
+        }
+    }
+}
+
+#[derive(Debug, Clone, Deserialize)]
+#[serde(rename_all = "kebab-case")]
+pub enum ProxyConfigType {
+    Socks5,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct ProxyConfig {
+    #[serde(rename = "type")]
+    pub proxy_type: ProxyConfigType,
+    pub host: String,
+    pub port: u16,
+    #[serde(default)]
+    pub username: String,
+    #[serde(default)]
+    pub password: String,
+}
+
+impl Into<Proxy> for ProxyConfig {
+    fn into(self) -> Proxy {
+        match self.proxy_type {
+            ProxyConfigType::Socks5 => irc::connection::Proxy::Socks5 {
+                host: self.host,
+                port: self.port,
+                username: self.username,
+                password: self.password,
+            },
         }
     }
 }
```

Comment:
We should always implement From instead of Into. Into will then be automatically derived.

Reply:
I don't know if it's a good practice to use the `From`  trait. 

Proper location for the implementation of `From` trait  is `irc::connection` module, while the `irc` crate is not dependent to the `data` crate.

the workaround would be to implement that inside `data::config::proxy` module which might be bad practice to separate the `irc::connection::Proxy` implementations in two separated locations.

Reply:
I'm not saying to relocate anything, keep this but just make it:

```rs
impl From<ProxyConfig> for Proxy
```

Instead of Into. You can do the from implementation and it _gives_ you into for free

Reply:
I've addressed in efd7b54